src/APRSsharp/Program.cs

```diff
@@ -35,19 +38,28 @@ public static void Main(string[] args)
             callsign = Console.ReadLine();
             Console.WriteLine("Enter your password: ");
             password = Console.ReadLine();
+            Console.WriteLine("Press key Q or q to cancel the connection:\n ");
             string? callsignArg = string.IsNullOrEmpty(callsign) ? null : callsign;
             string? passwordArg = string.IsNullOrEmpty(password) ? null : password;
             n.ReceivedTcpMessage += PrintTcpMessage;
-            n.Receive(callsignArg, passwordArg);
+            Task receive = n.Receive(callsignArg, passwordArg);
 
-            // skeleton method that will be used to handle the decoded packets
-            Console.WriteLine("Enter the packet name ");
-            var packetName = Console.ReadLine();
-            Packet p = new Packet();
-            p.DecodeInformationField(packetName);
-            Timestamp? ts = p.Timestamp;
-            Position? pos = p.Position;
-            Console.WriteLine($"\nHello, your packet name is, {p.Comment}, at cordinates, {pos?.Coordinates}, and time, {ts?.DateTime.Hour}");
+            ConsoleKeyInfo input;
+            input = Console.ReadKey();
+
+            while (true)
+            {
+                if (input.Key == ConsoleKey.Q)
+                {
+                    n.Disconnect();
+                    await receive;
+                    break;
+                }
+                else
+                {
+                   input = Console.ReadKey();
+                }
+            }
```

Comment:
This logic is absolutely correct. I'll recommend a tweak to make it more concise and avoid repeating the `input = Console.ReadKey();` line. When possible, we want to avoid repeating code. If there happens to be a bug in the code or the functionality needs to change, the developer implementing the fix may miss the second or third place the code is duplicated.

In this case, we can take advantage of a `while (true)` _always_ executing the first time and just put our `ReadKey` call at the top of the loop and any actions we take on it lower down.

```suggestion
            while (true)
            {
                ConsoleKeyInfo input = Console.ReadKey();

                if (input.Key == ConsoleKey.Q)
                {
                    n.Disconnect();
                    await receive;
                    break;
                }
            }
```